lib/web_ui/test/canvaskit/embedded_views_test.dart

```diff
@@ -910,8 +1155,92 @@ void testMain() {
       sb.addPlatformView(6, width: 10, height: 10);
       sb.pop();
       dispatcher.rasterizer!.draw(sb.build().layerTree);
-      expect(countCanvases(), 1);
+      expect(
+          sceneMatchesMarkers(<_EmbeddedViewMarker>[
+            overlay,
+            platformView,
+            platformView,
+            platformView,
+            platformView,
+            platformView,
+            overlay,
+          ]),
+          isTrue);
+
+      sb = LayerSceneBuilder();
+      sb.pushOffset(0, 0);
+      sb.addPlatformView(1, width: 10, height: 10);
+      sb.addPlatformView(2, width: 10, height: 10);
+      sb.addPlatformView(3, width: 10, height: 10);
+      sb.addPlatformView(4, width: 10, height: 10);
+      sb.pop();
+      dispatcher.rasterizer!.draw(sb.build().layerTree);
+      expect(
+          sceneMatchesMarkers(<_EmbeddedViewMarker>[
+            overlay,
+            platformView,
+            platformView,
+            platformView,
+            platformView,
+            overlay,
+          ]),
+          isTrue);
+
+      sb = LayerSceneBuilder();
+      sb.pushOffset(0, 0);
+      sb.addPlatformView(4, width: 10, height: 10);
+      sb.addPlatformView(3, width: 10, height: 10);
+      sb.addPlatformView(2, width: 10, height: 10);
+      sb.addPlatformView(1, width: 10, height: 10);
+      sb.pop();
+      dispatcher.rasterizer!.draw(sb.build().layerTree);
+      expect(
+          sceneMatchesMarkers(<_EmbeddedViewMarker>[
+            overlay,
+            platformView,
+            platformView,
+            platformView,
+            platformView,
+            overlay,
+          ]),
+          isTrue);
     });
     // TODO(dit): https://github.com/flutter/flutter/issues/60040
   }, skip: isIosSafari);
 }
+
+// Used to test that the platform views and overlays are in the correct order in
+// the scene.
+enum _EmbeddedViewMarker {
+  overlay,
+  platformView,
+}
+
+_EmbeddedViewMarker get overlay => _EmbeddedViewMarker.overlay;
+_EmbeddedViewMarker get platformView => _EmbeddedViewMarker.platformView;
+
+// TODO(hterkelsen): Make this into a Matcher for better error messages.
+bool sceneMatchesMarkers(List<_EmbeddedViewMarker> markers) {
+  final List<html.Element> sceneElements = flutterViewEmbedder
+      .sceneElement!.children
+      .where((html.Element element) => element.tagName != 'svg')
+      .toList();
+  if (markers.length != sceneElements.length) {
+    return false;
```

Comment:
If you moved the `expect()` statement here, you'd get a readable error message without having to implement a `Matcher`, e.g.:

```dart
expect(markers, hasLength(sceneElements.length));
```

Reply:
Done.